src/org/zaproxy/zap/extension/pscanrulesAlpha/BlankLinkTargetScanner.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Copyright 2018 The ZAP development team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+package org.zaproxy.zap.extension.pscanrulesAlpha;
+
+import java.util.List;
+
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.pscan.PassiveScanThread;
+import org.zaproxy.zap.extension.pscan.PluginPassiveScanner;
+
+import net.htmlparser.jericho.Element;
+import net.htmlparser.jericho.HTMLElementName;
+import net.htmlparser.jericho.Source;
+
+public class BlankLinkTargetScanner extends PluginPassiveScanner {
+
+    private static final String MESSAGE_PREFIX = "pscanalpha.blanktarget.";
+
+    private static final String REL_ATTRIBUTE = "rel";
+    private static final String TARGET_ATTRIBUTE = "target";
+    private static final String _BLANK = "_blank";
+    private static final String NOOPENER = "noopener";
+
+    private PassiveScanThread parent = null;
+
+    @Override
+    public void setParent(PassiveScanThread parent) {
+        this.parent = parent;
+    }
+
+    @Override
+    public void scanHttpRequestSend(HttpMessage msg, int id) {
+    }
+
+    @Override
+    public int getPluginId() {
+        return 10108;
+    }
+
+    @Override
+    public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
+        if (msg.getResponseBody().length() == 0 || !msg.getResponseHeader().isHtml()) {
+            // No point attempting to parse non-HTML content, it will not be correctly interpreted.
+            return;
+        }
+
+        List<Element> links = source.getAllElements(HTMLElementName.A);
+
+        for (Element link : links) {
+            // get target, check if its _blank
+            if (_BLANK.equals(link.getAttributeValue(TARGET_ATTRIBUTE))) {
+                // Not looking good,
+                if (!NOOPENER.equals(link.getAttributeValue(REL_ATTRIBUTE))) {
```

Comment:
Contains? It might have several keywords.

Reply:
We should also check for `noreferrer`, which implies `noopener`.

Reply:
Ref: https://www.w3.org/TR/html5/links.html#link-type-noreferrer
Also covers the `<area>` tag as mentioned earlier.